tests: Unittests don't build

#### Description
Unittests don't build in current master.
Quick bisect we did with @miri64 points to #8844 as introducing this problem.

#### Steps to reproduce the issue
```
cd tests/unittests
make
```

 or

```
make tests-<module>
```

#### Expected results
It should build.

#### Actual results
It does not build. Sample error:

```
make: *** No rule to make target '/home/vagrant/RIOT/tests/unittests/bin/samr21-xpro/tests-tweetnacl.a', needed by '/home/vagrant/RIOT/tests/unittests/bin/samr21-xpro/tests_unittests.elf'.  Stop.

```

#### Versions
Issue detected on Linux and OS X

--- a/Makefile.include
+++ b/Makefile.include
@@ -352,10 +352,13 @@ endif # RIOTNOLINK
 $(ELFFILE): $(BASELIBS)
 	$(Q)$(_LINK) -o $@
 
-# All modules are built by application.inc.mk makefile
 $(BINDIR)/$(APPLICATION_MODULE).a: $(RIOTBUILD_CONFIG_HEADER_C) $(USEPKG:%=$(BINDIR)/%.a) $(APPDEPS)
 	$(Q)DIRS="$(DIRS)" "$(MAKE)" -C $(APPDIR) -f $(RIOTMAKE)/application.inc.mk
 $(BINDIR)/$(APPLICATION_MODULE).a: FORCE
+
+# Other modules are built by application.inc.mk and packages building
+_SUBMAKE_LIBS = $(filter-out $(BINDIR)/$(APPLICATION_MODULE).a $(USEPKG:%=$(BINDIR)/%.a) $(APPDEPS), $(BASELIBS))
+$(_SUBMAKE_LIBS): $(BINDIR)/$(APPLICATION_MODULE).a $(USEPKG:%=$(BINDIR)/%.a)
 
 # 'print-size' triggers a rebuild. Use 'info-buildsize' if you do not need to rebuild.
 print-size: $(ELFFILE)

